webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/ImageValue/ImageValuePresentation.tsx

```diff
@@ -191,59 +181,60 @@ export const ImageValuePresentation: TabContainerPanelComponent<IDataValuePanelP
         cellValue: computed,
         canUpload: computed,
         src: computed,
-        savedSrc: computed,
+        cacheSrc: computed,
         canSave: computed,
         truncated: computed,
         model: observable.ref,
         resultIndex: observable.ref,
-        save: action.bound,
+        download: action.bound,
         upload: action.bound,
+        loadFullImage: action.bound,
       },
       { model, resultIndex, notificationService },
     );
 
-    const save = data.canSave ? data.save : undefined;
-    const upload = data.canUpload ? data.upload : undefined;
     const loading = model.isLoading();
-    const value = data.cellValue;
-
-    const load = async () => {
-      if (!data.selectedCell) {
-        return;
-      }
 
-      try {
-        await data.contentAction.resolveFileDataUrl(data.selectedCell);
-      } catch (exception: any) {
-        notificationService.logException(exception, 'data_viewer_presentation_value_content_download_error');
-      }
-    };
-
-    const valueSize = bytesToSize(isResultSetContentValue(value) ? value.contentLength ?? 0 : 0);
-    const isDownloadable = data.selectedCell && data.contentAction.isDownloadable(data.selectedCell);
+    const valueSize = bytesToSize(isResultSetContentValue(data.cellValue) ? data.cellValue.contentLength ?? 0 : 0);
+    const isTruncatedMessageDisplay = data.truncated && !data.src;
+    const isDownloadable = isTruncatedMessageDisplay && data.selectedCell && data.contentAction.isDownloadable(data.selectedCell);
+    const isCacheDownloading =
```

Comment:
this part is not quite obvious how it works. may be it is better to move this logic into `ResultSetCacheAction`

so each time when you download the cache we have a ready to go flag here

Reply:
fixed